File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        //counting sort:
        int cnt0 = 0, cnt1 = 0, cnt2 = 0;
        for(int n:nums){
            if(n == 0)
                cnt0++;
            else if(n == 1)
                cnt1++;
            else
                cnt2++;
        }
        int i = 0;
        while(cnt0 != 0){
            nums[i] = 0;
            i++;
            cnt0--;
        }
        while(cnt1 != 0){
            nums[i] = 1;
            i++;
            cnt1--;
        }
        while(cnt2 != 0){
            nums[i] = 2;
            i++;
            cnt2--;
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int cnt0 = 0, cnt1 = 0, cnt2 = 0;
        for(auto num:nums){
            if(num == 0)
                cnt0++;
            else if(num == 1)
                cnt1++;
            else
                cnt2++;
        }
        nums.clear();
        while(cnt0--){
            nums.push_back(0);
        }
        while(cnt1--){
            nums.push_back(1);
        }
        while(cnt2--){
            nums.push_back(2);
        }
    }
};